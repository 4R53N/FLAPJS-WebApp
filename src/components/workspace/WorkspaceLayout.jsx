import React from 'react';
import PropTypes from 'prop-types';
import Style from './WorkspaceLayout.module.css';

/**
 * A React component that separates into 2 layers: foreground and background.
 * This is usally used with other layouts to provide an overlay over a
 * background, which will usually contain the editor playground.
 */
class WorkspaceLayout extends React.Component
{
    constructor(props)
    {
        super(props);
    }
    
    /** @override */
    render()
    {
        const props = this.props;

        const renderBackground = props.renderBackground;
        
        return (
            <section ref={this.container}
                className={Style.container + ' ' + (props.className || '')}>
                <div className={Style.foreground}>
                    {props.children}
                </div>
                <div className={Style.background}>
                    {renderBackground && renderBackground(this)}
                </div>
            </section>
        );
    }
}

WorkspaceLayout.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    renderBackground: PropTypes.func,
};
WorkspaceLayout.defaultProps = {
};

export default WorkspaceLayout;
