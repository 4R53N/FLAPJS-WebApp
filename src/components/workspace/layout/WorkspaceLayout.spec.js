import React from 'react';
import { shallow } from 'enzyme';

import WorkspaceLayout from './WorkspaceLayout.jsx';

let wrapper;

beforeEach(() =>
{
    wrapper = shallow(<WorkspaceLayout/>);
});

describe('<WorkspaceLayout/> rendering', () =>
{
    test('renders correctly', () =>
    {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<WorkspaceLayout/> interaction', () =>
{
    /** I promise I will test you soon... */
});

describe('<WorkspaceLayout/> lifecycle', () =>
{
    /** ...later. */
});
