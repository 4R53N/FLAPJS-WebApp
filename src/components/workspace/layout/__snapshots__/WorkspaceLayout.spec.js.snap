// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkspaceLayout/> rendering renders correctly 1`] = `
<section
  className="container "
>
  <div
    className="background"
  />
  <div
    className="foreground"
  />
</section>
`;
